### ISSUE TITLE
Proper synchronous GPIO pin pulse methods

### ISSUE THREAD

**Issue by NogginBops** [Pi4J/pi4j-v1#391]
Currently there is no proper way to call the `pulse()` method synchronously. There is a flag in [GpioPinImpl.java - pulse()](https://github.com/Pi4J/pi4j/blob/master/pi4j-core/src/main/java/com/pi4j/io/gpio/impl/GpioPinImpl.java) to say that the call should block. 
```java
pulse(long duration, PinState pulseState, boolean blocking, Callable<Void> callback)
```

But this generates a problem when the thread gets interrupted.
It throws a `RuntimeException` instead of a `InterruptedException`.
Because the methods normal use-case does not throw an exception it would be inconvenient to add a `throws` declaration to the method. 
But this makes a lot of sense when you want to call it synchronously where you expect to get `InterruptedException`s from blocking calls.
Shouldn't there be synchronous methods available that throw the appropriate exception?

**Comment by savageautomate** [comment]
*Housekeeping*

Pi4J v2 includes separate synchronous and asynchronous pulse methods.

**Comment by NogginBops** [comment]
This is still a problem.
It's still impossible to call `pulse( blocking = true )` and expect correct behavior because of the `RuntimeException`.
https://github.com/Pi4J/pi4j/blob/master/pi4j-core/src/main/java/com/pi4j/io/gpio/impl/GpioPinImpl.java#L351

This should be changed to an `InterruptedException`.

**Comment by savageautomate** [comment]
**Fixed for v1.3.**  Added a compliment of `pulseSync()` methods to digital pin outputs.  Deprecated `pulse()` methods that contain the `blocking` parameter.  This way we don't break the interface for existing implementations.  

https://github.com/Pi4J/pi4j/blob/b9630767db601e74ef94534017862f0f13688199/pi4j-core/src/main/java/com/pi4j/io/gpio/GpioPinDigitalOutput.java#L73-L76


In an ideal world, I would prefer the `pulse()` methods to be synchronous and have separate `pulseAsync()` methods for the asynchronous variants.  But for legacy/backwards compatibility support we can't change this now.  This is the way it has been defined and implemented in Pi4J v2.x : 

https://github.com/Pi4J/pi4j-v2/blob/b1075269086a002b5c465868a1395ff8f9d318de/pi4j-core/src/main/java/com/pi4j/io/gpio/digital/DigitalOutput.java#L79-L96
